### PR TITLE
Seurat and SpatialExperiment integration

### PR DESCRIPTION
**Pull request by maddygriz**
counts, metadata, and coordinates are integrated. Images will take more thought and will come after image package

### CODE REVIEW

File: R/convertors.R [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+#' seurat_demoData <- to.Seurat(object = target_demoData, ident = "cell_line", 
+#'                              normData = "exprs_norm", forceRaw = FALSE)
+#' 
+#' @importFrom CreateSeuratObject Seurat
+#' @importFrom AddMetaData Seurat
+#' 
+#' @export 
+#' 
+
+to.Seurat <- function(object, ident, normData = "exprs_norm", coordinates = NULL, forceRaw = FALSE){
```
Comment (NicoleEO) [inline R/convertors.R:28]: objects are not guaranteed to have "exprs_norm". I would either default to "exprs" or set to NULL. If NULL add check.

File: R/convertors.R [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    if(object@featureType == "Probe"){
+        stop("Data must be on Target level before converting to a Seurat Object")
+    }
+    
+    if((length(names(object@assayData)) == 1 & names(object@assayData)[1] == "exprs") & forceRaw == FALSE){
+        stop("It is NOT recommended to use Seurat's normalization for GeoMx data. 
+             Normalize using GeomxTools::normalize() or set forceRaw to TRUE if you want to continue with Raw data")
+    }
+
```
Comment (NicoleEO) [inline R/convertors.R:37]: Can you add a check for the assay data name existing?

File: R/convertors.R [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+#' 
+#' seurat_demoData <- to.SpatialExperiment(object = target_demoData, normData = "exprs_norm", 
+#'                                         forceRaw = FALSE)
+#' 
+#' @importFrom SpatialExperiment SpatialExperiment
+#' 
+#' @export 
+#' 
+
+to.SpatialExperiment <- function(object, normData = "exprs_norm", coordinates = NULL, forceRaw = FALSE){
```
Comment (NicoleEO) [inline R/convertors.R:116]: Same here, either default to "exprs" or set to NULL. If NULL add check.

File: R/convertors.R [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    
+    if(object@featureType == "Probe"){
+        stop("Data must be on Target level before converting to a Seurat Object")
+    }
+    
+    if((length(names(object@assayData)) == 1 & names(object@assayData)[1] == "exprs") & forceRaw == FALSE){
+        stop("It is NOT recommended to use Seurat's normalization for GeoMx data. 
+             Normalize using GeomxTools::normalize() or set forceRaw to TRUE if you want to continue with Raw data")
+    }
+
```
Comment (NicoleEO) [inline R/convertors.R:126]: Add check for assay data name being present.

**Review by NicoleEO — CHANGES_REQUESTED** [review]
Can you increment the version and update news? I would like to make sure we increment every version a user can potentially pull even if from dev.
Please add test cases, update reqs and specs in qb as well so we can run test case before merge. Only need to include for the piece written so far.

**Comment by NicoleEO** [comment]
Can you also add update the authors in the description to match the readme?

**Review by NicoleEO — CHANGES_REQUESTED** [review]
A few minor edits. And I am also getting the following errors in the test.
image

**Review by NicoleEO — CHANGES_REQUESTED** [review]

File: R/coercions.R [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            stop(paste0("ident \"", ident, "\" not found in GeoMxSet Object"))
+        }
+        
+        Seurat::Idents(seuratConvert) <- seuratConvert[[ident]]
+    }
+    
+    
+    seuratConvert@misc <- object@experimentData@other 
+    seuratConvert@misc[["sequencingMetrics"]] <- sData(object)[colnames(sData(object)) %in% sequencingMetrics]
+    seuratConvert@misc[["QCMetrics"]] <- sData(object)[colnames(sData(object)) %in% QCMetrics]
```
[the remark this answers is not in the document]
Comment (maddygriz) [reply R/coercions.R:81]: The empty dataframe is now NULL. Tests have been updated. same for SpatialExperiment

**Comment by maddygriz** [comment]
Changes have all been added. Working on creating issues

**Review by NicoleEO — APPROVED** [review]
looks great

**Pull request merged**